chemicaltoolbox/mordred/mordred_descriptors.xml

```diff
@@ -0,0 +1,67 @@
+<tool id="ctb_mordred_descriptors" name="Calculate molecular descriptors" version="0.1.0">
+    <description>with Mordred</description>
+    <requirements>
+        <requirement type="package" version="1.1.2">mordred</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        python '$__tool_directory__/mordred_descriptors.py'
+            -i '${input}'
+            --iformat '${input.ext}' 
+            -o '${output}'
+            $header
+            $use_3d
+    ]]></command>
+    <inputs>
+        <param name="input" format="smi,sdf,inchi" type="data" label="Molecule data"
+            help="SDF, SMILES or InChi format"/>
+        <param name='header' type='boolean' truevalue='--header' falsevalue=''
+            label='Include a header line' help='Include names of the descriptors as the first line in the output file'/>
+        <param name='use_3d' type='boolean' truevalue='--3d' falsevalue=''
+            label='Calculate 3D descriptors' help='Include 3D as well as 2D descriptors - only valid if an SD-file is selected'/>
+    </inputs>
+    <outputs>
+        <data name="output" format="tabular" />
+    </outputs>
+    <tests>
+        <test>
+            <param name="input" ftype='sdf' value="10mol.sdf" />
+            <param name="header" value="False" />
+            <param name="use_3d" value="True" />
+            <output name="output" ftype='tabular' file="10sdf.tab" />
+        </test>
+        <test>
+            <param name="input" ftype='smi' value="8mol.smi" />
+            <param name="header" value="False" />
+            <param name="use_3d" value="False" />
+            <output name="output" ftype='tabular' file="8smi.tab" />
+        </test>
+    </tests>
+
+    <help><![CDATA[
+Calculates up to 1825 molecular descriptors using the Mordred package. A list of all
+descriptors is located here_.
+
+.. _here: https://static-content.springer.com/esm/art%3A10.1186%2Fs13321-018-0258-y/MediaObjects/13321_2018_258_MOESM3_ESM.xlsx
```

Comment:
Can we dump this into github somewhere and link to it?

Reply:
this okay? https://github.com/simonbray/mordred-descriptors